### PR TITLE
HPCC-13042 Disable replicate where needed

### PR DESCRIPTION
**Pull request by miguelvazq**
Disable replicate option within spray options and remote copy within logical files when not available.

Signed-off-by: Miguel Vazquez miguel.vazquez@lexisnexis.com

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

File: esp/src/eclwatch/DFUQueryWidget.js [hunk 1]
```diff
@@ -394,6 +399,27 @@ define([
...
         },
 
+        checkReplicate: function (value, checkBoxValue) {
+            var context = this;
+            WsTopology.TpGroupQuery({
+                request: {}
+            }).then(function (response) {
+                if (lang.exists("TpGroupQueryResponse.TpGroups.TpGroup", response)) {
+                    var arr = response.TpGroupQueryResponse.TpGroups.TpGroup;
+                    for (index in arr) {
```
Comment (GordonSmith) [inline esp/src/eclwatch/DFUQueryWidget.js]: missing "var" for index:  for (var index in arr)

File: esp/src/eclwatch/LZBrowseWidget.js [hunk 2]
```diff
@@ -468,6 +473,46 @@ define([
...
+        },
+
+        checkReplicate: function (value, checkBoxValue) {
+            var context = this;
+            FileSpray.GetSprayTargets({
+                request: {}
+            }).then(function (response) {
+                if (lang.exists("GetSprayTargetsResponse.GroupNodes.GroupNode", response)) {
+                    var arr = response.GetSprayTargetsResponse.GroupNodes.GroupNode;
+                    for (index in arr) {
```
Comment (GordonSmith) [inline esp/src/eclwatch/LZBrowseWidget.js]: Missing "var" (index just ends up as a global variable without it.

**Comment by GordonSmith** [comment]
Two minor comments otherwise looks fine.

**Comment by miguelvazq** [comment]
Updated, should be good now.

File: esp/src/Visualization [hunk 3]
```diff
@@ -1 +1 @@
-Subproject commit da4fdbc3eedec680f916d71c3f2d414730185a8b
+Subproject commit 28418fa21158e692a4cd49dde72996d8d04641bf
```
Comment (GordonSmith) [inline esp/src/Visualization]: You should not be changing the Visualization Submodule.

File: esp/src/eclwatch/DFUQueryWidget.js [hunk 4]
```diff
@@ -394,6 +399,27 @@ define([
             pMenu.startup();
         },
 
+        checkReplicate: function (value, checkBoxValue) {
+            var context = this;
```
Comment (GordonSmith) [inline esp/src/eclwatch/DFUQueryWidget.js]: context not used.

File: esp/src/eclwatch/LZBrowseWidget.js [hunk 5]
```diff
@@ -468,6 +473,46 @@ define([
...
+                context.checkReplicate(value, context.variableSprayReplicateCheckbox);
+            });
+
+            this.sprayBlobDestinationSelect.on('change', function (value){
+                context.checkReplicate(value, context.blobSprayReplicateCheckbox);
+            });
+        },
+
+        checkReplicate: function (value, checkBoxValue) {
+            var context = this;
```
Comment (GordonSmith) [inline esp/src/eclwatch/LZBrowseWidget.js]: context not used

**Comment by GordonSmith** [comment]
The visualization change needs to be resolved - the others are optional...

**Comment by miguelvazq** [comment]
Moved PR to https://github.com/hpcc-systems/HPCC-Platform/pull/7078

**Pull request closed without merging**